pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -571,6 +579,65 @@ func (a Adapter) generateDeploymentObjectMeta(deployment *appsv1.Deployment, lab
 	}
 }
 
+// deleteRemoteResourcesNotPresentInDevfile compares the list of Devfile K8s component and remote K8s resources
+// and removes the remote resources not present in the Devfile;
+// it ignores the core components (such as deployments, svc, pods; all resources with `component:<something>` label)
+func (a Adapter) deleteRemoteResourcesNotPresentInDevfile(labels map[string]string) error {
+	allRemoteK8sResources, err := a.kubeClient.GetAllResourcesFromSelector(util.ConvertLabelsToSelector(labels), a.kubeClient.GetCurrentNamespace())
```

Comment:
`labels` has been built with `odolabels.GetLabels`, which includes informative labels like `managed-by-version` one.

You will need to use `odolabels.GetSelector` instead to have a wider selection.

Reply:
Also, you can use `selector` instead of `labels` as param name, to differentiate label / selector.